Tests/PublisherTests/DefaultPublisher/DefaultPublisherTests.swift

```diff
@@ -353,6 +353,92 @@ class DefaultPublisherTests: XCTestCase {
         wait(for: [expectation], timeout: 5.0)
     }
     
+    func testAdd_whenTrackableWithSameIdIsCurrentlyBeingAdded_itWaitsForTheFirstAddToComplete_andWhenTheFirstAddSucceeds_theSecondAddSucceedsToo() {
+        var numberOfTimesConnectCalled: Int32 = 0
+        var connectCompletion: ResultHandler<Void>? = nil
+        let connectCalledExpectation = expectation(description: "ablyPublisher’s connect method is called")
+        ablyPublisher.connectCompletionHandler = { completion in
+            OSAtomicIncrement32(&numberOfTimesConnectCalled)
```

Comment:
Hmm, seems like `OSAtomicIncrement32` has been deprecated since iOS 10:
```
@available(iOS, deprecated: 10.0, message: "Use atomic_fetch_add_explicit(memory_order_relaxed) from <stdatomic.h> instead")
``` 
I'm not a fan of using code that's been deprecated, especially for so long. 
Seems like using the suggested `atomic_fetch_add_explicit(memory_order_relaxed)` would be a real pain though.

There's however an Apple developed atomics package for Swift, maybe we could switch to it instead?
https://github.com/apple/swift-atomics

Adding a new dependency just to use atomics in two tests seems like an overkill though, so I'll leave it up to you.

Reply:
Yeah, I'd prefer to not add another package unless we need it in more than 2 places. I've just switched it to use a serial queue for synchronisation instead.